src/mds/Locker.cc

```diff
@@ -3928,9 +3939,21 @@ bool Locker::_do_cap_update(CInode *in, Capability *cap,
   // xattrs update?
   if (xattr) {
     dout(7) << " xattrs v" << pi.inode->xattr_version << " -> " << m->head.xattr_version << dendl;
-    pi.inode->xattr_version = m->head.xattr_version;
-    auto p = m->xattrbl.cbegin();
-    decode_noshare(*pi.xattrs, p);
+    __u32 total = 0;
+    for (const auto &p : *pi.xattrs)
```

Comment:
good candidate for a helper perhaps?

Reply:
I mean this block of change and the one in `_do_snap_update` is pretty much same, so, just move the size calculation check into a helper (in this class itself).